HOMEWORK_30/FinalProject/Game.cpp

```diff
@@ -0,0 +1,435 @@
+
+    player.setTexture(playerTexture);
+    player.setScale(window.getSize().x * 0.1f / playerTexture.getSize().x,
+        window.getSize().y * 0.1f / playerTexture.getSize().y);
+    player.setPosition(window.getSize().x / 2.0f - player.getGlobalBounds().width / 2.0f,
+        window.getSize().y - player.getGlobalBounds().height - 10);
+
+    background.setTexture(backgroundTexture);
+    background.setScale(window.getSize().x / static_cast<float>(backgroundTexture.getSize().x),
+        window.getSize().y / static_cast<float>(backgroundTexture.getSize().y));
+
+    score = 0;
+    lastPlatformY = window.getSize().y;
+
+    scoreText.setFont(font);
+    scoreText.setCharacterSize(24);
+    scoreText.setFillColor(sf::Color::Black);
+    scoreText.setPosition(10, 40);
+
+    highScoreText.setFont(font);
+    highScoreText.setCharacterSize(24);
+    highScoreText.setFillColor(sf::Color::Black);
+    highScoreText.setPosition(10, 10);
+
+    respawnPlatforms();
+}
+
+void Game::resetGame() 
+{
+    state = GameState::Menu;
+    menuState.reset();
+    pauseState.reset();
+    gameOverState.reset();
+    initialize();
+}
+
+void Game::updateHighScore() 
+{
+    if (score > highScore) 
+    {
+        highScore = score;
+        writeHighScore(highScore);
+    }
+}
+
+void Game::respawnPlatforms() 
+{
+    platforms.clear();
+
+    float lastY = window.getSize().y - 50;
+    while (lastY > 0) 
+    {
+        std::shared_ptr<Platform> platform = generatePlatform(lastY);
+        platforms.push_back(platform);
+        lastY -= rand() % static_cast<int>(maxHeightGap - minHeightGap + 1) + minHeightGap;
+    }
+}
+
+std::shared_ptr<Platform> Game::generatePlatform(float yPos, bool isBroken, bool isMoving) 
+{
+    std::shared_ptr<Platform> platform;
+
+    if (isBroken) 
+    {
+        platform = std::make_shared<BrokenPlatform>();
+        platform->sprite.setTexture(brokenPlatformTexture);
+    }
+    else if (isMoving) 
+    {
+        platform = std::make_shared<MovingPlatform>(rand() % 4 + 2);
+        platform->sprite.setTexture(movingPlatformTexture);
+    }
+    else 
+    {
+        platform = std::make_shared<RegularPlatform>();
+        platform->sprite.setTexture(platformTexture);
+    }
+
+    platform->sprite.setScale(window.getSize().x * 0.25f / platform->sprite.getTexture()->getSize().x,
+        window.getSize().y * 0.03f / platform->sprite.getTexture()->getSize().y);
+
+    do 
+    {
+        platform->x = rand() % static_cast<int>(window.getSize().x - platform->sprite.getGlobalBounds().width);
+        platform->y = yPos;
+    } while (Platform::checkOverlap(platform, platforms));
+
+    platform->sprite.setPosition(platform->x, platform->y);
+    return platform;
+}
+
+void Game::handleEvents() 
+{
+    sf::Event event;
+    while (window.pollEvent(event)) 
+    {
+        if (event.type == sf::Event::Closed) 
+        {
+            window.close();
+        }
+
+        if (state == GameState::Menu) 
```

Comment:
each of these state handling code could be at least its own function. And ideally its own class